slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -218,6 +223,69 @@ std::string SLAMServiceImpl::PaintMap() {
     return jpeg_img;
 }
 
+bool SLAMServiceImpl::ExtractPointCloudToBuffer(std::stringbuf &buffer) {
+    std::ostream oss(&buffer);
+    bool has_points = false;
+
+    cartographer::mapping::MapById<cartographer::mapping::NodeId,
+                                   cartographer::mapping::TrajectoryNode>
+        trajectory_nodes;
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        trajectory_nodes =
+            map_builder.map_builder_->pose_graph()->GetTrajectoryNodes();
+    }
+
+    long number_points = 0;
+    for (auto trajectory_node : trajectory_nodes) {
+        auto point_cloud = trajectory_node.data.constant_data
+                               ->filtered_gravity_aligned_point_cloud;
+        number_points += point_cloud.size();
+    }
+
+    // Write our PCD file, which is written as a binary.
+    oss << "VERSION .7\n"
+        << "FIELDS x y z rgb\n"
+        << "SIZE 4 4 4 4\n"
+        << "TYPE F F F I\n"
+        << "COUNT 1 1 1 1\n"
+        << "WIDTH " << number_points << "\n"
+        << "HEIGHT " << 1 << "\n"
+        << "VIEWPOINT 0 0 0 1 0 0 0\n"
+        << "POINTS " << number_points << "\n"
+        << "DATA binary\n";
+
+    for (auto trajectory_node : trajectory_nodes) {
+        cartographer::sensor::PointCloud local_gravity_aligned_point_cloud =
+            trajectory_node.data.constant_data
+                ->filtered_gravity_aligned_point_cloud;
+
+        // We're only applying the transformation of the `global_pose` on the
```

Comment:
Is "transformation" supposed to be "translation" here?

Reply:
yes, thanks for catching this!